Feature/auth selection

SajeeTechi opened the pull request:
sign in with email
sign in with passkey
sign in with local account

blockchain managed auth sessions

2FA account factory

SajeeTechi committed:
update config: update config of wallet.pay.brussels to include session attribute

SajeeTechi committed:
remove deprecated env

SajeeTechi committed:
auth page with email signin option

SajeeTechi committed:
install shadcn form

SajeeTechi committed:
email form

SajeeTechi committed:
update @citizenwallet/sdk

SajeeTechi committed:
util functions for cw session

SajeeTechi committed:
email form submission action
- create request body

SajeeTechi committed:
base url of cw session project

SajeeTechi committed:
rename env

SajeeTechi committed:
remove logs

SajeeTechi committed:
- submit email form to get challenge
- verify session request tx hash for success

SajeeTechi committed:
shadcn input otp

SajeeTechi committed:
minor

SajeeTechi committed:
disable email field when submitting

SajeeTechi committed:
otp input page and form

SajeeTechi committed:
minor

SajeeTechi committed:
prettier formatting config

SajeeTechi committed:
prettier format code

SajeeTechi committed:
back navigation from /signin/email/otp

SajeeTechi committed:
zustand state for session

SajeeTechi committed:
local storage keys

SajeeTechi committed:
use function from cw sdk to await for tx success

SajeeTechi committed:
save variables in state and get account address

SajeeTechi committed:
modify accessing session state

SajeeTechi committed:
submit otp form
confirm session

SajeeTechi committed:
navigate to /address after session confirm success

SajeeTechi committed:
'isSessionExpired' function

SajeeTechi committed:
comments

SajeeTechi committed:
fix build errors

SajeeTechi committed:
prettier formatting

SajeeTechi committed:
Merge pull request #10 from citizenwallet/feature/auth-selection-email

Feature/auth selection email

SajeeTechi committed:
passkey compatibility check

SajeeTechi committed:
env for passkeys

SajeeTechi committed:
integrate passkey registration and authentication

SajeeTechi committed:
new local storage keys

SajeeTechi committed:
- signin with passkey and create account address
- require PR: https://github.com/citizenwallet/js-sdk/pull/4

SajeeTechi committed:
prettier format

SajeeTechi committed:
Merge pull request #11 from citizenwallet/feature/auth-selection-passkey

Feature/auth selection passkey

SajeeTechi committed:
router replace

SajeeTechi committed:
lowercase keys of local storage

SajeeTechi committed:
offer signin with local account

SajeeTechi committed:
prettier format

SajeeTechi committed:
Merge pull request #12 from citizenwallet/feature-auth-selection/local-signin

Feature auth selection/local signin

SajeeTechi committed:
move signin to root

SajeeTechi committed:
hook to determine account address and auth method

SajeeTechi committed:
offer sign-in options in root
- if account addrss exists, then navigate to /${accountAddress}

SajeeTechi committed:
hook to determine signin-in method and account address

SajeeTechi committed:
deeplink signin with v3 local format

SajeeTechi committed:
prettier formatting

SajeeTechi committed:
Merge pull request #13 from citizenwallet/feature-auth-selection/move-signin-to-root

Feature auth selection/move signin to root

SajeeTechi committed:
eval read-only mode for local  sign-in

SajeeTechi committed:
remove wallet container

SajeeTechi committed:
remove dev tools middleware

SajeeTechi committed:
re-initialise session store

SajeeTechi committed:
-reinit session store
-isSessionExpired
-getTwoAccountAddress

SajeeTechi committed:
eval session expiration for email and passkey signins

SajeeTechi committed:
redirect logic

SajeeTechi committed:
prettier formatting

SajeeTechi committed:
Merge pull request #14 from citizenwallet/feature-auth-selection/eval-read-write

Feature auth selection/eval read write

SajeeTechi committed:
remove unnecessary

SajeeTechi committed:
page to signout of auth method

SajeeTechi committed:
corection

SajeeTechi committed:
delete key function

SajeeTechi committed:
correction

SajeeTechi committed:
signout for auth methods email and passkeys
- clear state
- clear local storage

SajeeTechi committed:
logout of local account

SajeeTechi committed:
prettier format

SajeeTechi committed:
Merge pull request #15 from citizenwallet/feature-auth-selection/logout

Feature auth selection/logout

SajeeTechi committed:
remove signout page

SajeeTechi committed:
relocate signout to '/profile/[address]'

SajeeTechi committed:
edit styling of signout button

SajeeTechi committed:
Merge pull request #16 from citizenwallet/feature-auth-selection/relocate-signout

Feature auth selection/relocate signout

SajeeTechi committed:
disable send button when send is in progress

SajeeTechi committed:
todo

SajeeTechi committed:
todo

SajeeTechi committed:
resolve account address from 0x or username

SajeeTechi committed:
local account for session based and local accounts

SajeeTechi committed:
server action to resolve account address from username or ENS domain

SajeeTechi committed:
Merge pull request #17 from citizenwallet/feature-auth-selection/resolve-0x-or-username

Feature auth selection/resolve 0x or username

SajeeTechi committed:
root loading state

SajeeTechi committed:
clear state on signout

SajeeTechi committed:
use session util functions from sdk

SajeeTechi committed:
prettier format

SajeeTechi committed:
Merge branch 'main' into feature/auth-selection

SajeeTechi committed:
prettier format

SajeeTechi committed:
adapt change of generateConnectionMessage in local signin

SajeeTechi committed:
TODO

SajeeTechi committed:
new key for storage

SajeeTechi committed:
remove log

SajeeTechi committed:
bugfix infinite render

SajeeTechi committed:
save deeplink to handle later

SajeeTechi committed:
handle deeplink on mount

SajeeTechi committed:
restore communities.local from main

SajeeTechi committed:
test for parsing QR code and data

SajeeTechi committed:
handle sendTo QR for on deeplink

SajeeTechi commented:
Requires: https://github.com/citizenwallet/js-sdk/pull/10

SajeeTechi committed:
add session manager to some configs

SajeeTechi committed:
update community configs with session manager info

SajeeTechi committed:
update package @citizenwallet/sdk

kevtechi commented on .env.example:
```diff
@@ -14,4 +14,10 @@ NEXT_PUBLIC_REOWN_PROJECT_ID="x"
+# Passkey
+RELYING_PARTY_NAME='Citizen Wallet'
+RELYING_PARTY_ID='localhost' # domain without protocol or port
+RELYING_PARTY_ORIGIN='http://localhost:3000' # expected URL from which registration or authentication occurs
```
the wallet can be deployed on multiple domains at once, make this dynamic

SajeeTechi replied:
comment resolved with commit
https://github.com/citizenwallet/webapp/pull/19/commits/80a42f78ce8f78b57cdf51efe86c2d792768d2a2

SajeeTechi committed:
refactor(auth): make relying party ID dynamic based on origin domain

BREAKING CHANGE: Remove RELYING_PARTY_* environment variables

- Remove static RELYING_PARTY_* environment variables to support multi-domain deployments
- Add webauthn.ts utility to extract relying party ID from origin
- Add unit tests for relying party ID extraction
- Modify components to pass relying party values as props:
  - Update page-client.tsx to use getBaseUrl for origin
  - Update signin-passkey.tsx to accept relying party props
  - Update actions.ts to accept relying party values as arguments

This change enables the webapp to be deployed across multiple domains
without requiring environment variable configuration for each domain.

SajeeTechi commented:
Pending
Copy URL of local account before deletion

SajeeTechi committed:
feat(auth): local account signout UX

- Remove conditional text based on auth method for sign out button
- Keep special handling for local accounts with save URL dialog
- Enable users to copy and save local account URL before signing out
- Improve UX by clearly explaining the consequences of local account deletion

SajeeTechi committed:
bugfix: voucher URLs were making local accounts on deeplink
create local account on deeplink only if format matches

kevtechi committed:
bump communities with cw-safe paymasters

kevtechi commented on src/hooks/signin-method.ts:
```diff
@@ -0,0 +1,169 @@
+
+export type AuthMethod = "passkey" | "local" | "email" | "none";
+
+export function useSigninMethod(config: cwSDK.Config) {
```
we have state management set up in this repo with zustand. follow the same pattern

kevtechi commented on src/hooks/signin-method.ts:
```diff
@@ -0,0 +1,169 @@
+import { StorageService, StorageKeys } from "@/services/storage";
+import { CWAccount } from "@/services/account";
+import { getBaseUrl } from "@/utils/deeplink";
+import * as cwSDK from "@citizenwallet/sdk";
```
don't import like this. import the pieces you need from the sdk. better code splitting

kevtechi commented on test/qr.test.ts:
```diff
@@ -0,0 +1,161 @@
+import { parseQRFormat, QRFormat, parseQRCode } from "@citizenwallet/sdk"; // adjust import path as needed
+
+const testCases = [
```
not sure why we are writing test cases for the sdk here. put this in the sdk

kevtechi commented on src/containers/wallet/SendModal.tsx:
```diff
@@ -82,10 +85,14 @@ export default function SendModal({
   ) => {
     if (!resolvedTo) return;
+
+    setIsSending(true);
```
can use state management for this. the action is already there

SajeeTechi committed:
replace importing namespace

SajeeTechi committed:
move test file to js-sdk

SajeeTechi committed:
replace useState with state store attributes

SajeeTechi committed:
evaluate auth session in session store

kevtechi commented on src/hooks/signin-method.ts:
```diff
@@ -0,0 +1,169 @@
+type AuthMethod = "passkey" | "local" | "email" | "none";
+
+export function useSigninMethod(config: cwConfig) {
+  const [authMethod, setAuthMethod] = useState<AuthMethod>("none");
```
this whole file should not be needed anymore if this is all in state management

SajeeTechi committed:
state attributes for evaluating auth session

SajeeTechi committed:
evaluate auth session and return early if possible

SajeeTechi committed:
use session action to eval auth session

SajeeTechi committed:
use session action to eval auth session

SajeeTechi committed:
use session action to eval auth session

SajeeTechi committed:
deprecate sign-in hook

SajeeTechi committed:
backup modal only visible to local account type

SajeeTechi committed:
create local account url correctly
